src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -0,0 +1,693 @@
+end
+
+/-- The Hausdorff edistance is invariant under eisometries -/
+lemma Hausdorff_edist_image (h : isometry Φ) :
+  Hausdorff_edist (Φ '' s) (Φ '' t) = Hausdorff_edist s t :=
+begin
+  unfold Hausdorff_edist,
+  congr,
+  { ext b,
+    split,
+    { assume hb,
+      rcases (mem_image _ _ _ ).1 hb with ⟨y, ⟨hy, hy'⟩⟩,
+      rcases (mem_image _ _ _ ).1 hy with ⟨z, ⟨hz, hz'⟩⟩,
+      rw [← hy', ← hz', inf_edist_image h],
+      exact mem_image_of_mem _ hz },
+    { assume hb,
+      rcases (mem_image _ _ _ ).1 hb with ⟨y, ⟨hy, hy'⟩⟩,
+      rw [← hy', ← inf_edist_image h],
+      exact mem_image_of_mem _ (mem_image_of_mem _ hy) }},
+  { ext b,
+    split,
+    { assume hb,
+      rcases (mem_image _ _ _ ).1 hb with ⟨y, ⟨hy, hy'⟩⟩,
+      rcases (mem_image _ _ _ ).1 hy with ⟨z, ⟨hz, hz'⟩⟩,
+      rw [← hy', ← hz', inf_edist_image h],
+      exact mem_image_of_mem _ hz },
+    { assume hb,
+      rcases (mem_image _ _ _ ).1 hb with ⟨y, ⟨hy, hy'⟩⟩,
+      rw [← hy', ← inf_edist_image h],
+      exact mem_image_of_mem _ (mem_image_of_mem _ hy) }}
+end
+
+/-- The Hausdorff distance is controlled by the diameter of the union -/
+lemma Hausdorff_edist_le_ediam (hs : s ≠ ∅) (ht : t ≠ ∅) : Hausdorff_edist s t ≤ diam (s ∪ t) :=
+begin
+  rcases ne_empty_iff_exists_mem.1 hs with ⟨x, xs⟩,
+  rcases ne_empty_iff_exists_mem.1 ht with ⟨y, yt⟩,
+  refine Hausdorff_edist_le_of_mem_edist _ _,
+  { exact λz hz, ⟨y, yt, edist_le_diam_of_mem (subset_union_left _ _ hz) (subset_union_right _ _ yt)⟩ },
+  { exact λz hz, ⟨x, xs, edist_le_diam_of_mem (subset_union_right _ _ hz) (subset_union_left _ _ xs)⟩ }
+end
+
+/-- The Hausdorff distance satisfies the triangular inequality -/
+lemma Hausdorff_edist_triangle : Hausdorff_edist s u ≤ Hausdorff_edist s t + Hausdorff_edist t u :=
+begin
+  change Sup ((λx, inf_edist x u) '' s) ⊔ Sup ((λx, inf_edist x s) '' u) ≤ Hausdorff_edist s t + Hausdorff_edist t u,
+  simp only [and_imp, set.mem_image, lattice.Sup_le_iff, exists_imp_distrib,
+             lattice.sup_le_iff, -mem_image, set.ball_image_iff],
+  split,
+  show ∀x ∈ s, inf_edist x u ≤ Hausdorff_edist s t + Hausdorff_edist t u, from λx xs, calc
+    inf_edist x u ≤ inf_edist x t + Hausdorff_edist t u : inf_edist_le_inf_edist_add_Hausdorff_edist
+    ... ≤ Hausdorff_edist s t + Hausdorff_edist t u :
+      add_le_add_right' (inf_edist_le_Hausdorff_edist_of_mem  xs),
+  show ∀x ∈ u, inf_edist x s ≤ Hausdorff_edist s t + Hausdorff_edist t u, from λx xu, calc
+    inf_edist x s ≤ inf_edist x t + Hausdorff_edist t s : inf_edist_le_inf_edist_add_Hausdorff_edist
+    ... ≤ Hausdorff_edist u t + Hausdorff_edist t s :
+      add_le_add_right' (inf_edist_le_Hausdorff_edist_of_mem xu)
+    ... = Hausdorff_edist s t + Hausdorff_edist t u : by simp [Hausdorff_edist_comm, add_comm]
+end
+
+/-- The Hausdorff edistance between a set and its closure vanishes -/
+@[simp] lemma Hausdorff_edist_self_closure : Hausdorff_edist s (closure s) = 0 :=
```

Comment:
Similar to Chris's comment below, couldn't we deduce this from `Hausdorff_edist_closure₂` and `Hausdorff_edist_self`?

Reply:
This lemma is used in the proof of `Hausdorff_edist_closure₁` (in the `simp` invocation at the last line) so no, the order can not be changed.